java/core/src/main/java/sleeper/core/util/PollWithRetries.java

```diff
@@ -37,25 +41,56 @@ private PollWithRetries(long pollIntervalMillis, int maxPolls) {
         this.maxPolls = maxPolls;
     }
 
+    /**
+     * Creates an instance of this class.
+     *
+     * @param  pollIntervalMillis the poll interval in milliseconds
```

Comment:
Please can we show how the interval will be used? It's amount of time to wait for in between polls. This also applies to the static constructor that takes Duration objects.